Provide clear description of sidecar-injector in README (#39)
cyberark/sidecar-injector: README.md

# CyberArk Sidecar Injector

> Sidecars are used in Kubernetes to introduce additional features to application pods. Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector** enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). This means when **CyberArk Sidecar Injector** is deployed and enabled in a namespace, any pod created in that namespace, with annotations that indicate a request for injection, will result in automated sidecar injection. **CyberArk Sidecar Injector** provides a selection of available sidecars that are configurable through annotations.

Reviewer: ```suggestion
Sidecars are used in Kubernetes to introduce additional features to application pods. Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector** enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). This means that when **CyberArk Sidecar Injector** is deployed and enabled in a namespace, any pod created in that namespace with annotations that indicate a request for injection will result in automated sidecar injection. **CyberArk Sidecar Injector** provides a selection of available sidecars that are configurable through annotations with <insert names> being the currently-supported ones.
```

Reviewer: non-blocking nit: This paragraph for sure (and maybe even other ones in this doc) may need word wrapping.

> Sidecars are used in Kubernetes to introduce additional features to application pods. Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector** enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). This means when **CyberArk Sidecar Injector** is deployed and enabled in a namespace, any pod created in that namespace, with annotations that indicate a request for injection, will result in automated sidecar injection. **CyberArk Sidecar Injector** provides a selection of available sidecars that are configurable through annotations.

Reviewer 2: ```suggestion
Sidecars are used in Kubernetes to introduce additional features to application pods.
Manual sidecar injection can be cumbersome and repetitive. **CyberArk Sidecar Injector**
enables automatic sidecar injection through it being a [mutating admission webhook controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook).
This means when **CyberArk Sidecar Injector** is deployed and enabled in a namespace,
any pod created in that namespace with the appropriate annotations will result in automated
sidecar injection. **CyberArk Sidecar Injector** provides support for a selection of available sidecars
that are configurable through annotations.
```
shortens line length, makes minor edits

Author: updated

> Note that unlike manual injection, automatic injection occurs at the pod-level. You won’t see any change to the deployment itself. Instead you’ll want to check individual pods (via kubectl describe) to see the injected sidecar.

Reviewer: not a blocker but I think we italicize notes like this usually

> Note that unlike manual injection, automatic injection occurs at the pod-level. You won’t see any change to the deployment itself. Instead you’ll want to check individual pods (via kubectl describe) to see the injected sidecar.

Reviewer: ```suggestion
Note that unlike manual injection, automatic injection occurs at the pod-level. You won’t see any change to the deployment itself. Instead you will want to check individual pods (via `kubectl describe`) to see the injected sidecar.
```

***

> This document shows how to deploy and use the CyberArk Sidecar Injector [mutating admission webhook controller](https://kubernetes.io/docs/admin/admission-controllers/#mutatingadmissionwebhook-beta-in-19) which injects sidecar container(s) into a pod prior to persistence of the underlying object.

Reviewer: Is there a better link than a link to the beta page?

Author: yup updated
